docker-compose.yml

```diff
@@ -15,7 +15,7 @@ services:
       - 8003:8001
     volumes:
       - ./src:/usr/src/app:z
-    command: "python manage.py runserver 0.0.0.0:8001"
+    command: "python -m uvicorn officehours.asgi:application --host=0.0.0.0 --port=8001 --reload"
```

Comment:
I decided to bump to `4.0.0` for Django Channels, since the issue that was blocking us seems to have been resolved in `channels_redis` 4.1.0. In Channels 4, they decoupled the implementation from the `daphne` 4.0.0. You can still use `daphne` for development, but it's a separate package and it needs to be added to `APPS` in Django. I figured rather than having two ASGI server implementations installed, we may as well use `uvicorn` for local development. It has a `--reload` feature and looks to work well, with really just some logging changes that I can see. I don't think there will be logging changes to production since we're already using `uvicorn` -- unless there was a change in `uvicorn` itself picked up by the version change.

Reply:
This sounds good, I've switched to gunicorn for development for other projects that aren't upgraded to use uvicorn yet.